x-pack/plugins/alerting/server/task_runner/task_runner.ts

```diff
@@ -422,7 +460,14 @@ export class TaskRunner<
         )
       );
     } else {
-      this.logger.debug(`no scheduling of actions for alert ${alertLabel}: alert is muted.`);
+      if (muteAll) {
+        this.logger.debug(`no scheduling of actions for alert ${alertLabel}: alert is muted.`);
+      }
+      if (!this.shouldLogAndScheduleActionsForAlerts()) {
+        this.logger.debug(
+          `no scheduling of actions for alert ${alertLabel}: alert execution has been cancelled.`
```

Comment:
Do we have access to a cancellation reason here? If so, is it helpful in this message? I know we only have one use case for cancel now but if we add more in the future, perhaps it's helpful to indicate why the execution cancelled?

Reply:
We may have a use for a "cancellation reason" in the shutdown case - https://github.com/elastic/kibana/issues/117513 ; reason: kibana shutting down

And you could imagine, some day, wanting to explicitly cancel some super-long-running rule, explicitly; reason: user action

So ya, I think we will need it.

Reply:
Is it ok to create a followup issue for this? Since `cancel` is initiated by task manager, I'm thinking the cancellation reason would originate from there and changing the signature of the cancel function to pass in a reason might affect other task types.

Reply:
Sure, sounds good. It'll help the o11y effort in the future if we can have more context into why things happened.

Reply:
Kinda good to defer anyway, since we don't **need** it right now, and by the time we do, our thinking on the shape may change ...

Reply:
Created https://github.com/elastic/kibana/issues/118544